src/main/java/it/gov/pagopa/payhub/activities/connector/classification/AssessmentsRegistryServiceImpl.java

```diff
@@ -32,4 +36,25 @@ public void createAssessmentsRegistryByDebtPositionDTOAndIudList(DebtPositionDTO
             .build(),
         authnService.getAccessToken());
   }
+
+  @Override
+  public void createAssessmentsRegistry(AssessmentsRegistry assessmentsRegistry) {
+    assessmentsRegistryClient.createAssessmentsRegistry(assessmentsRegistry, authnService.getAccessToken());
+  }
+
+  @Override
+  public Optional<AssessmentsRegistry> searchAssessmentsRegistryByBusinessKey(
+          Long organizationId, String debtPositionTypeOrgCode, String sectionCode, String officeCode, String assessmentCode, String operatingYear) {
+
+    PagedModelAssessmentsRegistry pagedModelAssessmentsRegistry = assessmentsRegistryClient.getAssessmentsRegistry(
+                    organizationId, debtPositionTypeOrgCode, sectionCode, officeCode, assessmentCode, operatingYear,
+                    authnService.getAccessToken(), 0, 1, null);
+
+    var embedded = pagedModelAssessmentsRegistry.getEmbedded();
+      if (embedded == null || embedded.getAssessmentsRegistries() == null || embedded.getAssessmentsRegistries().isEmpty()) {
```

Comment:
use org.springframework.util.CollectionUtils#isEmpty(java.util.Collection<?>) instead
```suggestion
      if (embedded == null || CollectionUtils.isEmpty(embedded.getAssessmentsRegistries()) {
```

Reply:
Fixed